File: Strings/__1000__/1287/1287.js
const inputs = require("fs").readFileSync("/dev/stdin", "utf8").split("\n");
const valor_maximo = 2147483647;

inputs.pop(); // Remove a última linha vazia

for (const input of inputs) {
    let num = "";
    let is_valid = true;

    for (let i = 0; i < input.length; i++) {
        const ch = input[i];

        if (ch >= '0' && ch <= '9') {
            num += ch;
        } else if (ch === 'o' || ch === 'O') {
            num += '0';
        } else if (ch === 'l') {
            num += '1';
        } else if (ch === ' ' || ch === ',') {
            continue;
        } else {
            is_valid = false;
            break;
        }
    }

    const erro = num === "" || Number(num) > valor_maximo;
    console.log(is_valid && !erro ? String(Number(num)) : "error");
}

File: Strings/__1000__/1287/1287.py
#dado uma entrada precisa corrigir um erro fatal, para isso
#ignore espacos em branco e caracter A-Z a-z e caracteres
#especiais como vigula, pont e etc...

#se o caracter for "l" trate como 1
#se caracter for O ou o trata como 0
#nenhum valor deve ser maior que 2147483647
#se não tiver valor inteiro ou não satisfazer os requisitos
#print error

valor_maximo = 2147483647

while True:
    try:
        n = input()
        num = ""
        is_valid = True

        for char in n:
            if char.isdigit():
                num += char
            elif char in ['O', 'o']:
                num += '0'
            elif char == 'l':
                num += '1'
            elif char in [' ', ',']:
                continue
            else:
                is_valid = False
                break

        error = num == "" or int(num) > valor_maximo
        print(int(num) if (not error and is_valid) else "error")
    
    except EOFError:
        break